common/unsafe/src/test/java/org/apache/spark/unsafe/PlatformUtilSuite.java

```diff
@@ -157,4 +159,15 @@ public void heapMemoryReuse() {
     Assert.assertEquals(onheap4.size(), 1024 * 1024 + 7);
     Assert.assertEquals(obj3, onheap4.getBaseObject());
   }
+
+  @Test
+  // SPARK-26021
+  public void writeMinusZeroIsReplacedWithZero() {
+    byte[] doubleBytes = new byte[Double.BYTES];
+    byte[] floatBytes = new byte[Float.BYTES];
+    Platform.putDouble(doubleBytes, Platform.BYTE_ARRAY_OFFSET, -0.0d);
+    Platform.putFloat(floatBytes, Platform.BYTE_ARRAY_OFFSET, -0.0f);
+    Assert.assertEquals(0, Double.compare(0.0d, ByteBuffer.wrap(doubleBytes).getDouble()));
```

Comment:
are you sure this test fails before the fix? IIUC `0.0 == -0.0` is ture, but they have different binary format

Reply:
BTW thanks for adding the unit test! It's a good complementary to the end-to-end test.

Reply:
yeah, it fails. Indeed 0.0 == -0.0 so I'm using Double.compare == 0 to test this.